### DISCUSSION TITLE
Missing characters in WSL

### DISCUSSION THREAD

**Discussion by johanmattssonm** [starship/starship#3651] [Q&A] [answered]
Hi. I tried starship in the WSL with FiraCode NF. It looks like some characters are missing:
![image](https://user-images.githubusercontent.com/350392/155394245-3e34b198-ea47-4688-9720-7a5ea489b8ba.png)

Is there a way to fix that?

**Comment by chipbuster** [comment]
The missing symbol is a ❯ glyph (emoji? character?). Can you see it on the web interface? I've put it below on its own line for clarity:

```
❯
```

Also, what terminal are you using to access WSL?

**Comment by johanmattssonm** [comment, marked as answer]
Thank you for the swift reply. I installed Ubuntu on Windows 11 I don't know which terminal that is but I use the default.

**Reply by chipbuster** [reply]
Like that's just the default application when you search the start menu for the WSL?

**Reply by johanmattssonm** [reply]
Yes, it is the default terminal. I see it when I search for Ubuntu or WSL.

Here is what I see in preferences, it might give a clue:
![image](https://user-images.githubusercontent.com/350392/155403528-467a5ec2-242b-4b57-a8be-218ccf51e30a.png)

**Reply by chipbuster** [reply]
Gotcha. So this is probably using the Windows ConHost which is known to have issues with UTF-8. Could you try using [Windows Terminal](https://www.microsoft.com/en-us/p/windows-terminal/9n0dx20hk701#activetab=pivot:overviewtab)?

**Reply by johanmattssonm** [reply]
Thank you. It looks like it did the trick. Brilliant support. No need to fiddle with the font.

**Reply by johanmattssonm** [reply]
It looks like most characters work now but I still see one box:
![image](https://user-images.githubusercontent.com/350392/155784360-1837bb72-5323-4171-8deb-a92d1ba830f4.png)

Copy and pasting the missing character to a browser tells me that it is U+E0A0 (Private Use Area). 

What should be in that spot? Is there a special font that I need to use?
Cheers

**Reply by chipbuster** [reply]
Unless you've changed the defaults, I believe that should be the branch symbol from the powerline font set (see [here](https://github.com/ryanoasis/powerline-extra-symbols#glyphs)).

We may be changing how starship works in the near-ish future, but I don't know exactly when this wil happen. Here's an overview of your options right now:

  - Use a font with powerline support (I believe Microsoft's `Cascadia Code PL` font has powerline by default). Some (rare) symbols in starship will not work (e.g. Erlang or NodeJS) until #3544 lands.
  - Install and use a [NerdFont](https://www.nerdfonts.com). This is more complicated but will ensure that all symbols will display right now. It will become unnecessary in a future release of starship, but will not cause issues if you keep using it.
  - Find some other way to import powerline symbols (not recommended due to setup difficulty, but the most flexible)

**Comment by johanmattssonm** [comment]
I certainly volunteer to update a font with the missing glyphs if that is needed but I do need to know which font to use.
